js/source/tile.js

```diff
@@ -32,12 +32,12 @@ Tile.prototype = {
      * @returns {Object} position
      * @private
      */
-    positionAt: function(coord, sourceMaxZoom) {
-        coord = coord.zoomTo(Math.min(this.coord.z, sourceMaxZoom));
+    positionAt: function(coord) {
+        var zoomedCoord = coord.zoomTo(this.coord.z);
         return {
-            x: (coord.column - this.coord.x) * this.tileExtent,
-            y: (coord.row - this.coord.y) * this.tileExtent,
-            scale: this.scale
+            x: (zoomedCoord.column - this.coord.x) * this.tileExtent,
+            y: (zoomedCoord.row - this.coord.y) * this.tileExtent,
+            scale: this.tileSize
```

Comment:
This doesn't _seem_ right to me but matches the behavior before https://github.com/mapbox/mapbox-gl-js/commit/002c69071275997886fd98aa4dbc6025eed36a86, as verified by running the new unit tests on 6deae336507751992d4c88cd7c024d48a10e56cc